<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jean Morrison" />

<meta name="date" content="2019-07-11" />

<title>Introduction to CAUSE</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>




<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CAUSE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Introduction</a>
</li>
<li>
  <a href="ldl_cad.html">Software Tutorial</a>
</li>
<li>
  <a href="pipeline.html">Pipeline</a>
</li>
<li>
  <a href="explore.html">Explore Results</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jean997/cause">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Introduction to CAUSE</h1>
<h4 class="author">Jean Morrison</h4>
<h4 class="date">2019-07-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>cause/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjean997causetreef183b4c4bc07b5606fd59a6974075d23e2351de1targetblankf183b4ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jean997/cause/tree/f183b4c4bc07b5606fd59a6974075d23e2351de1" target="_blank">f183b4c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjean997causetreef183b4c4bc07b5606fd59a6974075d23e2351de1targetblankf183b4ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jean997/cause/tree/f183b4c4bc07b5606fd59a6974075d23e2351de1" target="_blank">f183b4c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ll_v7_notes.Rmd
    Ignored:    ll_v7_notes.html
    Ignored:    pipeline_code/ld/
    Ignored:    pipeline_code/plink_reference/
    Ignored:    pipeline_code/raw_data/
    Ignored:    sim_results/
    Ignored:    src/RcppExports.o
    Ignored:    src/cause.so
    Ignored:    src/log_likelihood_functions.o
    Ignored:    tests/

Untracked files:
    Untracked:  .Rhistory (tumble-track's conflicted copy 2021-07-15)
    Untracked:  analysis/mrcieu.Rmd
    Untracked:  cause.Rcheck/
    Untracked:  example_data/chr22_AF0.05_0.1.RDS
    Untracked:  example_data/chr22_AF0.05_snpdata.RDS

Unstaged changes:
    Deleted:    analysis/figure/simulations.Rmd/plot_fp-1.png
    Deleted:    analysis/figure/simulations.Rmd/roc-1.png
    Deleted:    analysis/figure/simulations.Rmd/unnamed-chunk-3-1.png
    Deleted:    analysis/figure/simulations.Rmd/unnamed-chunk-5-1.png
    Deleted:    analysis/figure/simulations.Rmd/unnamed-chunk-6-1.png
    Modified:   src/RcppExports.cpp

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/f183b4c4bc07b5606fd59a6974075d23e2351de1/analysis/index.Rmd" target="_blank">f183b4c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2021-09-14
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/da5b784adce5647fa0d8d5ef8dc7dd8ecc6a92a4/docs/index.html" target="_blank">da5b784</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-11-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/c7a1cb42f2d86d4fe0eff4c19ef2a001c75eec28/analysis/index.Rmd" target="_blank">c7a1cb4</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-11-13
</td>
<td>
refactor website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/1673e5485c309669bc26e5a4097aeabb17fdf6de/docs/index.html" target="_blank">1673e54</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/ecffe6dbde2b122384ca39320e60c116a294fdc6/analysis/index.Rmd" target="_blank">ecffe6d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-02-09
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/70e2b97cc2ac9ecda4f2c3bc9dd1aaae04e00276/docs/index.html" target="_blank">70e2b97</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-12-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/a1c2e0978c9b46baa71f4f3909ab7ceb9cffa371/docs/index.html" target="_blank">a1c2e09</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-10-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/8bbb5162069dbefc21f0fca851275dcfed218a96/docs/index.html" target="_blank">8bbb516</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/86ffddd20efe2d2834dc5b682ada896f97c55bdd/analysis/index.Rmd" target="_blank">86ffddd</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-15
</td>
<td>
wflow_publish(c(“analysis/index.Rmd,” “analysis/simulations.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/350353625a9f789887406985e0f5f8d575829b7d/docs/index.html" target="_blank">3503536</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/8fbc50cf34adcd19c0b122f56b639f7c5e238003/analysis/index.Rmd" target="_blank">8fbc50c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-12
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/f3d82d3480c106f8e4b3febf0dd98dacfa7b9600/docs/index.html" target="_blank">f3d82d3</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/fdcf760dc0a55d9d4261088093a93d90d38939ce/analysis/index.Rmd" target="_blank">fdcf760</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-11
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/09acd92ce8c269e94f2e40536c29ffb50a74c9ac/docs/index.html" target="_blank">09acd92</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/a5de9b27fb757688681dd86913fe9ca953a7dbb0/analysis/index.Rmd" target="_blank">a5de9b2</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-11
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/5e4c977d829d49f92b3e7ef9cc82bdef80996887/docs/index.html" target="_blank">5e4c977</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/237d888154d88f8a04def84eb004736504fac0b2/analysis/index.Rmd" target="_blank">237d888</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-09
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/4a40efa8d7a279c09b16175f0ef7d1b42f0c1cf7/docs/index.html" target="_blank">4a40efa</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/6f137f1d6364b580bdb088844998e5ab73a2e077/analysis/index.Rmd" target="_blank">6f137f1</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-26
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/3b5c7e24db65f87044fb22570dc9331fdbf4601a/docs/index.html" target="_blank">3b5c7e2</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/a55827daf376aec55b6708d8797e67e62bb11fa3/analysis/index.Rmd" target="_blank">a55827d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
wflow_publish(files = c(“analysis/index.Rmd,” “analysis/ldl_cad.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/cb84ab212eff670b301173676a62b66cc4ee7ca3/docs/index.html" target="_blank">cb84ab2</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/a54a556ff8ac53d476f804f26088ddf95a9faac0/analysis/index.Rmd" target="_blank">a54a556</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
wflow_publish(files = c(“analysis/index.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/6b826d9bdb72220b0de8c9723f892ba31d7ddf0b/docs/index.html" target="_blank">6b826d9</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/286f4e9e0a432565b96c38bba3a1b155c02f9a78/docs/index.html" target="_blank">286f4e9</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/8f3b82e734b124b6d4d9d28b9defcce8d3726410/analysis/index.Rmd" target="_blank">8f3b82e</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
wflow_publish(files = c(“analysis/about.Rmd,” “analysis/index.Rmd,” “analysis/ldl_cad.Rmd,” “analysis/license.Rmd,”
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/24510f128068f566d4efca738364151712ceda03/analysis/index.Rmd" target="_blank">24510f1</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-22
</td>
<td>
small language changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/46b159a1f6313e26149b7fb021b2d18aaf4bbd5c/docs/index.html" target="_blank">46b159a</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/19edfa41426350036d7812260cbdba1ca9a7bcda/analysis/index.Rmd" target="_blank">19edfa4</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
wflow_git_commit(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/f15238b6e5b2e1f07f8846e5c56618a8ba28b62a/docs/index.html" target="_blank">f15238b</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/d163752b8091f97bcf76e4b6b012a2f23954bfd8/analysis/index.Rmd" target="_blank">d163752</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-14
</td>
<td>
update about and index
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/efc43ef89c766d22c038de5cb2b980fd1a7402c2/analysis/index.Rmd" target="_blank">efc43ef</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-14
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weclome-to-the-cause-website" class="section level1">
<h1>Weclome to the CAUSE website!</h1>
<p>Here are some useful links</p>
<p><a href="https://www.biorxiv.org/content/10.1101/682237v3">Pre-print</a></p>
<p><a href="https://github.com/jean997/cause">R package</a></p>
<p><a href="WNAR_2019.pdf">Slides!</a> from WNAR 2019</p>
<p>CAUSE is a Mendelian Randomization method using genome-wide summary statistics. CAUSE models correlated and uncorrelated horizontal pleiotropy in order to avoid false positives that can occur using other methods. Read a short introduction to the method below. In the tabs you can find a tutorial about using our software, some simulation results and an example of a larger data analysis.</p>
</div>
<div id="introduction-to-cause" class="section level1">
<h1>Introduction to CAUSE</h1>
<div id="summary-statistic-mendelian-randomizatoin-basics" class="section level2">
<h2>Summary Statistic Mendelian Randomizatoin Basics</h2>
<p>Mendelian randomization (MR) is a method for inferring causal effects from observational data by using genetic variants as insturmental variables. The key idea of MR is to treat genotypes as naturally occurring “randomizations” (<span class="citation">Smith and Hemani (2014)</span>,<span class="citation">Boef, Dekkers, and Le Cessie (2015)</span>,<span class="citation">Zheng et al. (2017)</span>). Suppose we are interested in the causal effect of trait <span class="math inline">\(M\)</span> (for “Mediator”) on trait <span class="math inline">\(Y\)</span>. The simplest MR methods assume that we can identify a genetic variant <span class="math inline">\(G_j\)</span>, that meets two assumptions:</p>
<ol style="list-style-type: decimal">
<li><span class="math inline">\(G_j\)</span> causally affects <span class="math inline">\(M\)</span></li>
<li><span class="math inline">\(G_j\)</span> has no affects on <span class="math inline">\(Y\)</span> that are not mediated through <span class="math inline">\(M\)</span></li>
</ol>
<p>These assumptions are illustrated in the following figure:</p>
<p><img src="cause_figure_1_standalone.png" style="width:30.0%" /></p>
<p>Here we have divided assumption 2 into two parts. First the variant may not affect any confounders/shared factors that act on both <span class="math inline">\(M\)</span> and <span class="math inline">\(Y\)</span> (correlated pleiotropy) and second, the variant cannot have any affects on <span class="math inline">\(Y\)</span> through a non-shared mechanism (uncorrelated pleiotropy).</p>
<p>If these assumptions hold thenthe associations of <span class="math inline">\(G_j\)</span> with traits <span class="math inline">\(M\)</span> and <span class="math inline">\(Y\)</span> will satisfy</p>
<p><span class="math display">\[
\beta_{Y,j} = \gamma \beta_{M,j},
\]</span> where <span class="math inline">\(\beta_{Y,j}\)</span> is the association of <span class="math inline">\(G_j\)</span> with <span class="math inline">\(Y\)</span>, <span class="math inline">\(\beta_{M,j}\)</span> is the association of <span class="math inline">\(G_j\)</span> with <span class="math inline">\(M\)</span> and <span class="math inline">\(\gamma\)</span> is the causal effect of <span class="math inline">\(M\)</span> on <span class="math inline">\(Y\)</span>. This relationship is the core of simple MR methods. Many methods based on Equation , including the commonly used inverse variance weighted (IVW) regression, first obtain estimates of <span class="math inline">\(\beta_{Y,j}\)</span> and <span class="math inline">\(\beta_{M,j}\)</span> for several genetic variants <span class="math inline">\(G_j\)</span>, and then estimate <span class="math inline">\(\gamma\)</span> by regressing the estimates of <span class="math inline">\(\beta_{Y,j}\)</span> on the estimates of <span class="math inline">\(\beta_{M,j}\)</span> (<span class="citation">Burgess, Dudbridge, and Thompson (2016)</span>).</p>
</div>
<div id="violating-mr-assumptions" class="section level2">
<h2>Violating MR Assumptions</h2>
<p>Correlated and uncorrelated pleiotropy are both forms of what has been previously termed horizontal pleiotropy. However, they have different effects on MR estimators. In uncorrelated pleiotropy, horizontal pleiotropic effects of variants are uncorrelated with effects on <span class="math inline">\(M\)</span>. This adds a noise term to the relationship above. On the other hand, when some variants exhibit correlated pleiotropy, this can induce an average correaltion between effect estimates even when the causal effect is equal to zero. This makes correlated pleiotropy more difficult to account for.</p>
<p>Several proposals have been made for accounting for horizontal pleiotropy in MR. However, most of these rely on the assumption that all horizontal pleiotropy is uncorrelated. These include Egger regression (<span class="citation">Bowden, Smith, and Burgess (2015)</span>,<span class="citation">Barfield et al. (2018)</span>) which adds an intercept term to the regression of <span class="math inline">\(\hat{\beta}_{Y,j}\)</span> on <span class="math inline">\(\hat{\beta}_{M,j}\)</span>, and several methods that rely on outlier removal including GSMR (<span class="citation">Zhu et al. (2018)</span>) and MR-PRESSO (<span class="citation">Verbanck et al. (2018)</span>). Another proposal, the weighted median estimator(<span class="citation">Bowden et al. (2016)</span>), makes no assumptions about the form of horizontal pleiotropy but assumes that fewer than half the variants exhibit horizontal pleiotropy.</p>
<p>The figures below demonstrate how correlated pleiotropy can lead to false positives using simple MR methods. On the left we have simulated data with no causal effect but 15% of the variants have correlated pleiotropic effecgts on <span class="math inline">\(Y\)</span> (blue triangles). Even though there are many variants with strong association with <span class="math inline">\(M\)</span> and no association with <span class="math inline">\(Y\)</span> (evidence against a causal effect) IVW regression obtains a <span class="math inline">\(p\)</span>-value of 0.01. We would like to be able to distinguish this scenario from the scenario on the right. On the right, we have simulated data with a true causal effect. Effect estimates are correlated for all varaints.</p>
<p><img src="scatter4_updlegend.png" style="width:48.0%" /> <img src="scatter5_updlegend.png" style="width:48.0%" /> The purpose of CASUE is to distinguish the patterns created by correlated pleiotropy from those created by a causal effect while still accounting for uncorrelated horizontal pleiotropy. We can also account for sample overlap in the GWAS of traits <span class="math inline">\(M\)</span> and <span class="math inline">\(Y\)</span> and we can avoid some other probelems encountered by simple MR by modeling uncertainty in effect sizes rather than pre-selecting variants with strong evidence of affecting trait <span class="math inline">\(M\)</span>.</p>
</div>
<div id="cause-model-overview" class="section level2">
<h2>CAUSE Model overview</h2>
<p>CAUSE uses a mixture model for variants included in the analysis. We assume that a proportion <span class="math inline">\(q\)</span> of variants exhibit correlated pleiotropy (the blue triangles in the figure above) while the rest do not. Correlated pleiotropic variants have the causal diagram on the right in the figure below. The remaining variants have the causal diagram on the left.</p>
<p><img src="cause_figure_3_standalone.png" style="width:50.0%" /></p>
<p>Note that we allow all variants to have uncorrelated pleiotropic effects (<span class="math inline">\(\theta_j\)</span>). To make the model identifiable and to encourage parsimoniuos solutions, we assume that <span class="math inline">\(q\)</span> is small. This assumption is encoded in a prior on <span class="math inline">\(q\)</span> that places most of its weight on small values (<span class="math inline">\(q\sim~\)</span>Beta<span class="math inline">\((1, 10)\)</span> by default). If <span class="math inline">\(Z_j\)</span> is an indicator that variant <span class="math inline">\(G_j\)</span> is a correlated pleiotropic variant then the model above implies that</p>
<p><span class="math display">\[
  \beta_{Y,j} = \underbrace{\gamma \beta_{M,j}}_{\text{causal effect}} + \underbrace{Z_j \eta \beta_{M,j}}_{\substack{\text{correlated}\\ \text{pleiotropy}}} + \underbrace{\theta_j}_{\substack{\text{uncorrelated}\\ \text{pleiotropy}}}.
\]</span> This relationship is the core idea of CAUSE. We estimate posterior distributions of <span class="math inline">\(\gamma\)</span>, <span class="math inline">\(\eta\)</span> and <span class="math inline">\(q\)</span> and compare the fit of posteriors from models with and without a causal effect. Our software provides posterior distribution estimates under two models: The sharing model in which <span class="math inline">\(\gamma\)</span> is fixed at 0 and the causal model, which allows <span class="math inline">\(\gamma\)</span> to be a free parameter. It also provides a test that the posteriors estimated under the causal model fit the data significantly better than posteriors estimated under the sharing model. If this is the case, we conclude that the data are consistent with a causal effect, or in other words, show the pattern of the right hand scatter plot above. For more details on model fitting and comparison, check out the paper! For details on running the software, look at the tutorial tab above.</p>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="unnumbered">References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Barfield2017" class="csl-entry">
Barfield, Richard, Helian Feng, Alexander Gusev, Lang Wu, and Wei Zheng. 2018. <span>“<span class="nocase">Transcriptome-wide association studies accounting for co- localization using Egger regression</span>.”</span> <em>Genetic Epidemiology</em> 42: 418–33.
</div>
<div id="ref-Boef2015" class="csl-entry">
Boef, Anna G. C., Olaf M. Dekkers, and Saskia Le Cessie. 2015. <span>“<span class="nocase">Mendelian randomization studies: A review of the approaches used and the quality of reporting</span>.”</span> <em>International Journal of Epidemiology</em> 44 (2): 496–511. <a href="https://doi.org/10.1093/ije/dyv071">https://doi.org/10.1093/ije/dyv071</a>.
</div>
<div id="ref-Bowden2016" class="csl-entry">
Bowden, Jack, M. Fabiola Del Greco, Cosetta Minelli, George Davey Smith, Nuala A. Sheehan, and John R. Thompson. 2016. <span>“<span class="nocase">Assessing the suitability of summary data for two-sample mendelian randomization analyses using MR-Egger regression: The role of the I2statistic</span>.”</span> <em>International Journal of Epidemiology</em> 45 (6): 1961–74. <a href="https://doi.org/10.1093/ije/dyw220">https://doi.org/10.1093/ije/dyw220</a>.
</div>
<div id="ref-Bowden2015" class="csl-entry">
Bowden, Jack, George Davey Smith, and Stephen Burgess. 2015. <span>“<span class="nocase">Mendelian Randomization Methodology Mendelian randomization with invalid instruments : effect estimation and bias detection through Egger regression</span>.”</span> <em>International Journal of Epidemiology</em>, no. June: 512–25. <a href="https://doi.org/10.1093/ije/dyv080">https://doi.org/10.1093/ije/dyv080</a>.
</div>
<div id="ref-Burgess2016" class="csl-entry">
Burgess, Stephen, Frank Dudbridge, and Simon G. Thompson. 2016. <span>“<span class="nocase">Combining information on multiple instrumental variables in Mendelian randomization: Comparison of allele score and summarized data methods</span>.”</span> <em>Statistics in Medicine</em> 35 (11): 1880–1906. <a href="https://doi.org/10.1002/sim.6835">https://doi.org/10.1002/sim.6835</a>.
</div>
<div id="ref-Smith2014" class="csl-entry">
Smith, George Davey, and Gibran Hemani. 2014. <span>“<span class="nocase">Mendelian randomization: genetic anchors for causal inference in epidemiological studies</span>.”</span> <em>Human Molecular Genetics</em> 23 (1): 89–98. <a href="https://doi.org/10.1093/hmg/ddu328">https://doi.org/10.1093/hmg/ddu328</a>.
</div>
<div id="ref-Verbanck2018" class="csl-entry">
Verbanck, Marie, Chia-yen Chen, Benjamin Neale, and Ron Do. 2018. <span>“<span class="nocase">Detection of widespread horizontal pleiotropy in causal relationships inferred from Mendelian randomization between complex traits and diseases</span>.”</span> <em>Nature Genetics</em> 50 (May). <a href="https://doi.org/10.1038/s41588-018-0099-7">https://doi.org/10.1038/s41588-018-0099-7</a>.
</div>
<div id="ref-Zheng2017" class="csl-entry">
Zheng, Jie, Denis Baird, Maria-Carolina Borges, Jack Bowden, Gibran Hemani, Philip Haycock, David M. Evans, and George Davey Smith. 2017. <span>“<span class="nocase">Recent Developments in Mendelian Randomization Studies</span>.”</span> <em>Current Epidemiology Reports</em> 4 (4): 330–45. <a href="https://doi.org/10.1007/s40471-017-0128-6">https://doi.org/10.1007/s40471-017-0128-6</a>.
</div>
<div id="ref-Zhu2018" class="csl-entry">
Zhu, Zhihong, Zhili Zheng, Futao Zhang, Yang Wu, Maciej Trzaskowski, Robert Maier, Matthew R. Robinson, et al. 2018. <span>“<span class="nocase">Causal associations between risk factors and common diseases inferred from GWAS summary data</span>.”</span> <em>Nature Communications</em> 9 (1): 224. <a href="https://doi.org/10.1038/s41467-017-02317-2">https://doi.org/10.1038/s41467-017-02317-2</a>.
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
