<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jean Morrison" />

<meta name="date" content="2018-10-16" />

<title>Example Analysis with CAUSE: LDL -&gt; CAD</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CAUSE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Introduction</a>
</li>
<li>
  <a href="ldl_cad.html">Software Tutorial</a>
</li>
<li>
  <a href="pipeline.html">Pipeline</a>
</li>
<li>
  <a href="explore.html">Explore Results</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jean997/cause">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Example Analysis with CAUSE: LDL -&gt; CAD</h1>
<h4 class="author">Jean Morrison</h4>
<h4 class="date">2018-10-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-11-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>cause/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20181014code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20181014)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20181014code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20181014)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjean997causetree7bdec64f86c8b4d694f5943dcdaac4f0c017c18ftargetblank7bdec64a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jean997/cause/tree/7bdec64f86c8b4d694f5943dcdaac4f0c017c18f" target="_blank">7bdec64</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjean997causetree7bdec64f86c8b4d694f5943dcdaac4f0c017c18ftargetblank7bdec64a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jean997/cause/tree/7bdec64f86c8b4d694f5943dcdaac4f0c017c18f" target="_blank">7bdec64</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ll_v7_notes.Rmd
    Ignored:    ll_v7_notes.html
    Ignored:    pipeline_code/ld/
    Ignored:    pipeline_code/plink_reference/
    Ignored:    pipeline_code/raw_data/
    Ignored:    sim_results/
    Ignored:    src/RcppExports.o
    Ignored:    src/cause.so
    Ignored:    src/log_likelihood_functions.o
    Ignored:    tests/

Untracked files:
    Untracked:  .Rhistory (tumble-track's conflicted copy 2021-07-15)
    Untracked:  cause.Rcheck/
    Untracked:  example_data/chr22_AF0.05_0.1.RDS
    Untracked:  example_data/chr22_AF0.05_snpdata.RDS
    Untracked:  example_data/ieu-b-38.vcf.gz
    Untracked:  example_data/ieu-b-38.vcf.gz.tbi
    Untracked:  example_data/ukb-b-9405.vcf.gz
    Untracked:  example_data/ukb-b-9405.vcf.gz.tbi

Unstaged changes:
    Deleted:    analysis/figure/simulations.Rmd/plot_fp-1.png
    Deleted:    analysis/figure/simulations.Rmd/roc-1.png
    Deleted:    analysis/figure/simulations.Rmd/unnamed-chunk-3-1.png
    Deleted:    analysis/figure/simulations.Rmd/unnamed-chunk-5-1.png
    Deleted:    analysis/figure/simulations.Rmd/unnamed-chunk-6-1.png
    Modified:   src/RcppExports.cpp

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/ldl_cad.Rmd</code>) and HTML (<code>docs/ldl_cad.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/7bdec64f86c8b4d694f5943dcdaac4f0c017c18f/analysis/ldl_cad.Rmd" target="_blank">7bdec64</a>
</td>
<td>
Jean Morrison
</td>
<td>
2021-11-29
</td>
<td>
wflow_publish(files = c(“analysis/mrcieu.Rmd”, “analysis/ldl_cad.Rmd”))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/d252cf60e08619c20363d5c5f0e1f04250a7d217/analysis/ldl_cad.Rmd" target="_blank">d252cf6</a>
</td>
<td>
Jean Morrison
</td>
<td>
2021-11-16
</td>
<td>
spelling errors
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/05b1882c3b4bb8ae03f0eab05b6ec880750a1247/docs/ldl_cad.html" target="_blank">05b1882</a>
</td>
<td>
Jean Morrison
</td>
<td>
2021-07-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/78949fc2e9967f8587db709e932f8db95f0fadab/analysis/ldl_cad.Rmd" target="_blank">78949fc</a>
</td>
<td>
Jean Morrison
</td>
<td>
2021-07-08
</td>
<td>
update tutorial
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/da5b784adce5647fa0d8d5ef8dc7dd8ecc6a92a4/docs/ldl_cad.html" target="_blank">da5b784</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-11-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/c7a1cb42f2d86d4fe0eff4c19ef2a001c75eec28/analysis/ldl_cad.Rmd" target="_blank">c7a1cb4</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-11-13
</td>
<td>
refactor website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/81a1a48e3401219d0fea62242133be7117dc8656/docs/ldl_cad.html" target="_blank">81a1a48</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/f8dff24eee9d6ea7ec83effa30317f2cc0240a1f/analysis/ldl_cad.Rmd" target="_blank">f8dff24</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-06-02
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/1cfcb0b3cc513064b0fecc385bb2317e482c6718/docs/ldl_cad.html" target="_blank">1cfcb0b</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/3c9571d068a1c2cdc2caa3472b90157de6357eea/analysis/ldl_cad.Rmd" target="_blank">3c9571d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-05-29
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/70e2b97cc2ac9ecda4f2c3bc9dd1aaae04e00276/docs/ldl_cad.html" target="_blank">70e2b97</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-12-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/b1c1567b304381e18e1b7c824630be7f38cbde69/analysis/ldl_cad.Rmd" target="_blank">b1c1567</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-08-07
</td>
<td>
update vignette to use new formatting code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/1a90dd649497a3c846f8abc2832de648ac881e80/docs/ldl_cad.html" target="_blank">1a90dd6</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/02aa566eb42e92315345923066a7fbdb9d921115/analysis/ldl_cad.Rmd" target="_blank">02aa566</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-15
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/1702accf8cc1746a5c599ab18b712450765ba676/docs/ldl_cad.html" target="_blank">1702acc</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/02bc6d3fcdc32e05b97ce870f6d3708c4e2d0e92/docs/ldl_cad.html" target="_blank">02bc6d3</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/28d211f5a11dd01b2b326d301864fc464cba1ba6/analysis/ldl_cad.Rmd" target="_blank">28d211f</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-09
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/bfa7d2888b1ddb77a0b565246f2ba6b7c0a7d007/docs/ldl_cad.html" target="_blank">bfa7d28</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/d8f89fc6d0b98cfb193685a5aa3eba2f92d5a3a2/analysis/ldl_cad.Rmd" target="_blank">d8f89fc</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-09
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/3b5c7e24db65f87044fb22570dc9331fdbf4601a/docs/ldl_cad.html" target="_blank">3b5c7e2</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/a55827daf376aec55b6708d8797e67e62bb11fa3/analysis/ldl_cad.Rmd" target="_blank">a55827d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
wflow_publish(files = c(“analysis/index.Rmd”, “analysis/ldl_cad.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/d8d148608a68c9fd853964a999d3ac3e216978cc/docs/ldl_cad.html" target="_blank">d8d1486</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/e0a6df4a672d4a6b6db130fe9f937792351cf567/analysis/ldl_cad.Rmd" target="_blank">e0a6df4</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
wflow_publish(files = c(“analysis/ldl_cad.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/33b373230107121d24eb71d1d4a7bdd70488f4c4/docs/ldl_cad.html" target="_blank">33b3732</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/0e352686c547156d6b4f7085d5779325717090c9/analysis/ldl_cad.Rmd" target="_blank">0e35268</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
wflow_publish(files = c(“analysis/ldl_cad.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/286f4e9e0a432565b96c38bba3a1b155c02f9a78/docs/ldl_cad.html" target="_blank">286f4e9</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/8f3b82e734b124b6d4d9d28b9defcce8d3726410/analysis/ldl_cad.Rmd" target="_blank">8f3b82e</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
<td>
wflow_publish(files = c(“analysis/about.Rmd”, “analysis/index.Rmd”, “analysis/ldl_cad.Rmd”, “analysis/license.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/1753b227154a949d80ae0ece3d13ef18f92f29b7/docs/ldl_cad.html" target="_blank">1753b22</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-09
</td>
<td>
update website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/2eb09d8a6a0e5cb986f1ffd6a71a6e44ca2567b9/docs/ldl_cad.html" target="_blank">2eb09d8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
<td>
added new option for a truncated prior on q
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/4a8f76ccd2554d1f46985507b41899ced9e02a3e/docs/ldl_cad.html" target="_blank">4a8f76c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/2652be12bb63e67611ef6eee0647254b790ba52a/analysis/ldl_cad.Rmd" target="_blank">2652be1</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/6ae7a60c30167592b1328cc1ee64958f29086ee1/docs/ldl_cad.html" target="_blank">6ae7a60</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-24
</td>
<td>
build website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/48bdf215b8c526c53cc9a0308677675ef231b2e9/analysis/ldl_cad.Rmd" target="_blank">48bdf21</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-24
</td>
<td>
fixing warnings
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/8c57c8a3cc8b0faa204a5dfa1a89336039622947/analysis/ldl_cad.Rmd" target="_blank">8c57c8a</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-24
</td>
<td>
build website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/8c57c8a3cc8b0faa204a5dfa1a89336039622947/docs/ldl_cad.html" target="_blank">8c57c8a</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-24
</td>
<td>
build website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/a34393dedf81051e37783951c9e3b74aba80de65/analysis/ldl_cad.Rmd" target="_blank">a34393d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-24
</td>
<td>
build website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/a34393dedf81051e37783951c9e3b74aba80de65/docs/ldl_cad.html" target="_blank">a34393d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-24
</td>
<td>
build website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/6354c3568e965503bbdde96d84c04c8801405bfd/docs/ldl_cad.html" target="_blank">6354c35</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/24510f128068f566d4efca738364151712ceda03/analysis/ldl_cad.Rmd" target="_blank">24510f1</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-22
</td>
<td>
small language changes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/d050da86b8956c154f24ef61a9c1ad270b7702b2/analysis/ldl_cad.Rmd" target="_blank">d050da8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-22
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/bbe49018f1c79ea1f2bfb1c1c4ac2228722dac60/docs/ldl_cad.html" target="_blank">bbe4901</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/558cd32e08be455b3a9cca07c0caf5515b3a8ca2/analysis/ldl_cad.Rmd" target="_blank">558cd32</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/73690eb831172fcd7130e9b4d2620f85b0c528c1/docs/ldl_cad.html" target="_blank">73690eb</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/1a891e3b8f5e85ae882d7b6b31fb0f590825c34c/analysis/ldl_cad.Rmd" target="_blank">1a891e3</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/d10191f6c1e324ef6073a95893546ccd9474bed1/analysis/ldl_cad.Rmd" target="_blank">d10191f</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
wflow_git_commit(“analysis/ldl_cad.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This document will walk through a real genome-sized example of how to use CAUSE. Some of the steps will take 5-10 minutes. For steps that require long computation we also provide output files that can be downloaded to make it easier to run through the example.</p>
<p>We will be analyzing GWAS data for LDL cholesterol and for coronary artery disease to test for a causal relationship of LDL on CAD. The analysis will have the following steps:</p>
<ol style="list-style-type: decimal">
<li>Format the data for use with CAUSE</li>
<li>Calculate nuisance parameters</li>
<li>LD pruning</li>
<li>Fit CAUSE</li>
<li>Look at results</li>
</ol>
<p>There are two ways to do the LD pruning in step 3. The easiest way is to use Plink which is the method we use here. There are also built-in functions in the <code>cause</code> package that allow you to do LD pruning with pre-computed pairwise LD files. This could let you use an alternate LD calculation. The last section of this document shows how to do that using LD information estimated from the 1000 Genomes CEU population using LDshrink <a href="https://zenodo.org/record/1464357#.W8a-fxROmV4">here</a>. These data are about 11 Gb.</p>
<p>The GWAS data we will use are about 320 Gb. However, in this tutorial you will be able to skip the large data steps and simply download the results.</p>
</div>
<div id="step-0-install-cause" class="section level2">
<h2>Step 0: Install CAUSE</h2>
<p>Follow installation instructions <a href="https://github.com/jean997/cause">here</a></p>
</div>
<div id="step-1-format-data-for-cause" class="section level2">
<h2>Step 1: Format Data for CAUSE</h2>
<p>This section describes formatting for data that are available in a flat file. If you have vcf files downloaded from the IEU Open GWAS project see <a href="mrcieu.html">here</a> for formatting instructions.</p>
<p>We will use <code>read_tsv</code> to read in summary statistics for a GWAS of LDL cholesterol and a GWAS of coronary artery disease from the internet. We will then combine these and format them for use with CAUSE. First read in the data. For LDL Cholesterol, we use summary statistics from Willer et al (2013) [PMID: 24097068]. For CAD we use summary statistics from van der Harst et al. (2017) [PMID: 29212778]. Downloading and formatting the data takes several minutes. If you want to skip this step, we provide a formatted data file that you can download below.</p>
<pre class="r"><code>library(readr)
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(cause)</code></pre>
<pre class="r"><code>X1 &lt;- read_tsv(&quot;http://csg.sph.umich.edu/abecasis/public/lipids2013/jointGwasMc_LDL.txt.gz&quot;)</code></pre>
<pre><code>Rows: 2437751 Columns: 10</code></pre>
<pre><code>── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (5): SNP_hg18, SNP_hg19, rsid, A1, A2
dbl (5): beta, se, N, P-value, Freq.A1.1000G.EUR</code></pre>
<pre><code>
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>X2 &lt;- read_tsv(&quot;ftp://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/vanderHarstP_29212778_GCST005194/CAD_META.gz&quot;)</code></pre>
<pre><code>Rows: 7947837 Columns: 18</code></pre>
<pre><code>── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr  (5): MarkerName, Allele1, Allele2, Direction, oldID
dbl (13): Freq1, FreqSE, MinFreq, MaxFreq, Effect, StdErr, P-value, HetISq, ...</code></pre>
<pre><code>
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<p>CAUSE needs the following information from each data set: SNP or variant ID, effect size, and standard error, effect allele and other allele. We provide a simple function that will merge data sets and produce a <code>cause_data</code> object that can be used with later functions. This step and the rest of the analysis are done in R.</p>
<p>The function <code>gwas_merge</code> will merge two data sets and and align effect sizes to correspond to the same allele. It will remove variants with ambiguous alleles (G/C or A/T) or with alleles that do not match between data sets (e.g A/G in one data set and A/C in the other). It will also remove variants that are duplicated in either data set. It will not remove variants that are simply strand flipped between the two data sets (e. g. A/C in one data set, T/G in the other). If p-values are available it will accept those but they are not required. If p-values are missing, be default they will be computed using a normal approximation but this can be bypassed by setting <code>compute_._pval = FALSE</code>. p-values are only used in the LD pruning step.</p>
<p>LDL column headers:</p>
<ul>
<li>SNP: rsid</li>
<li>Effect: beta</li>
<li>Standard Error: se</li>
<li>Effect Allele: A1</li>
<li>Other Allele: A2</li>
<li>p-value (optional): P-value</li>
</ul>
<p>CAD column headers:</p>
<ul>
<li>SNP: oldID</li>
<li>Effect: Effect</li>
<li>Standard Error: StdErr</li>
<li>Effect Allele: Allele1</li>
<li>Other Allele: Allele2</li>
<li>p-value (optional): P-value</li>
</ul>
<p>If the p-value column in either data set is missing, the <code>pval_cols</code> argument can be omitted or one of the elements can be NA.</p>
<pre class="r"><code>X &lt;- gwas_merge(X1, X2, snp_name_cols = c(&quot;rsid&quot;, &quot;oldID&quot;), 
                       beta_hat_cols = c(&quot;beta&quot;, &quot;Effect&quot;), 
                       se_cols = c(&quot;se&quot;, &quot;StdErr&quot;), 
                       A1_cols = c(&quot;A1&quot;, &quot;Allele1&quot;), 
                       A2_cols = c(&quot;A2&quot;, &quot;Allele2&quot;), 
                       pval_cols = c(&quot;P-value&quot;, &quot;P-value&quot;))</code></pre>
<pre><code>Formatting X1
There are  2437751  variants.
Removing  794  duplicated variants leaving  2436956 variants.
Removing  1  variants with illegal alleles leaving  2436956 variants.
Removed  375645  variants with ambiguous strand.
Flipping strand and effect allele so A1 is always A
Returning  2061311  variants.
Formatting X2
There are  7947837  variants.
Removing  13  duplicated variants leaving  7947811 variants.
Removing  1  variants with illegal alleles leaving  7947811 variants.
Removed  1202723  variants with ambiguous strand.
Flipping strand and effect allele so A1 is always A
Returning  6745088  variants.
After merging and removing variants with inconsistent alleles,  there are  2023354  variants that are present in both studies and can be used with CAUSE.</code></pre>
<p>Alternatively, you can download already formatted data <a href="https://github.com/jean997/cause/blob/master/example_data/LDL_CAD_merged.RDS">here</a> and read it in using <code>readRDS</code>.</p>
<pre class="r"><code>system(&quot;mkdir example_data/&quot;)
download.file(&quot;https://github.com/jean997/cause/blob/master/example_data/LDL_CAD_merged.RDS&quot;, destfile = &quot;example_data/LDL_CAD_merged.RDS&quot;)</code></pre>
<pre class="r"><code>X &lt;- readRDS(&quot;example_data/LDL_CAD_merged.RDS&quot;)</code></pre>
<pre class="r"><code>head(X)</code></pre>
<pre><code>        snp beta_hat_1   seb1      p1 beta_hat_2   seb2 A1 A2        p2
1 rs4747841     0.0037 0.0052 0.71580     0.0106 0.0056  A  G 0.0587000
2 rs4749917    -0.0033 0.0052 0.77480    -0.0108 0.0056  A  G 0.0538200
3  rs737656     0.0099 0.0054 0.04000     0.0196 0.0058  A  G 0.0007217
4  rs737657     0.0084 0.0054 0.08428     0.0195 0.0058  A  G 0.0007519
5 rs7086391    -0.0075 0.0067 0.26890     0.0115 0.0072  A  G 0.1088000
6  rs878177    -0.0073 0.0055 0.13760    -0.0225 0.0059  A  G 0.0001398</code></pre>
<p>There are likely more efficient ways to do this merge. If you would like to process the data yourself, you can construct a <code>cause_data</code> object from a data frame using the constructor <code>new_cause_data(X)</code> where <code>X</code> is any data frame that includes the columns <code>snp</code>, <code>beta_hat_1</code>, <code>seb1</code>, <code>beta_hat_2</code>, and <code>seb2</code>.</p>
</div>
<div id="step-2-calculate-nuisance-parameters" class="section level2">
<h2>Step 2: Calculate nuisance parameters</h2>
<p>The next step is to estimate the parameters that define the prior distribution of <span class="math inline">\(\beta_{M}\)</span> and <span class="math inline">\(\theta\)</span> and to estimate <span class="math inline">\(\rho\)</span>, the correlation between summary statistics that is due to sample overlap or population structure. We will do this with a random subset of 1,000,000 variants since our data set is large. <code>est_cause_params</code> estimates the nuisance parameters by finding the maximum a posteriori estimate of <span class="math inline">\(\rho\)</span> and the mixing parameters when <span class="math inline">\(\gamma = \eta = 0\)</span>. This step takes a several minutes.</p>
<pre class="r"><code>set.seed(100)
varlist &lt;- with(X, sample(snp, size=1000000, replace=FALSE))
params &lt;- est_cause_params(X, varlist)</code></pre>
<pre><code>Estimating CAUSE parameters with  1000000  variants.
1 0.1233775 
2 0.0005297492 
3 6.181563e-06 
4 4.140447e-08 </code></pre>
<p>The object <code>params</code> is of class <code>cause_params</code> and contains information about the fit as well as the maximum a posteriori estimates of the mixing parameters (<span class="math inline">\(\pi\)</span>) and <span class="math inline">\(\rho\)</span>. The object <code>params$mix_grid</code> is a data frame defining the distribution of summary statistics. The column <code>S1</code> is the variance for trait 1 (<span class="math inline">\(M\)</span>), the column <code>S2</code> is the variance for trait 2 (<span class="math inline">\(Y\)</span>) and the column <code>pi</code> is the mixture proportion assigned to that variance combination.</p>
<pre class="r"><code>class(params)</code></pre>
<pre><code>[1] &quot;cause_params&quot;</code></pre>
<pre class="r"><code>names(params)</code></pre>
<pre><code>[1] &quot;rho&quot;       &quot;pi&quot;        &quot;mix_grid&quot;  &quot;loglik&quot;    &quot;PIS&quot;       &quot;RHO&quot;      
[7] &quot;LLS&quot;       &quot;converged&quot; &quot;prior&quot;    </code></pre>
<pre class="r"><code>params$rho</code></pre>
<pre><code>[1] 0.06465784</code></pre>
<pre class="r"><code>head(params$mix_grid)</code></pre>
<pre><code>           S1          S2           pi
1 0.000000000 0.000000000 0.3105286679
2 0.000000000 0.003440730 0.2381082081
3 0.000000000 0.004865928 0.1490899384
4 0.003533615 0.004865928 0.1660234991
5 0.004997287 0.009731855 0.1177553305
6 0.019989147 0.009731855 0.0006521846</code></pre>
<p>In this case, we have estimated that 31% of variants have trait 1 variance and trait 2 equal to 0 meaning that they have no association with either trait.</p>
<p>Tip: Do not try to estimate the nuisance parameters with substantially fewer than 100,000 variants. This can lead to poor estimates of the mixing parameters whih leads to bad model comparisons.</p>
<p>If you don’t want to wait for this step, the parameters object can be downloaded from <a href="https://github.com/jean997/cause/blob/master/example_data/LDL_CAD_params.RDS">here</a> using</p>
<pre class="r"><code>download.file(&quot;https://github.com/jean997/cause/blob/master/example_data/LDL_CAD_params.RDS&quot;, destfile = &quot;example_data/LDL_CAD_params.RDS&quot;)
params &lt;- readRDS(&quot;example_data/LDL_CAD_params.RDS&quot;)</code></pre>
</div>
<div id="step-3-ld-pruning" class="section level2">
<h2>Step 3: LD Pruning</h2>
<p>We estimate CAUSE posterior distributions using an LD pruned set of variants, prioritizing variants with low trait <span class="math inline">\(M\)</span> (LDL) <span class="math inline">\(p\)</span>-values.</p>
<p>The easiest way is to use Plink to perform LD clumping. The <code>ieugwasr</code> package provides a convenient R interface to Plink. This method is fast but requires a reference sample which can be accessed through an API using <code>ieugwasr::ld_clump</code> (see help for that function). You can download also download reference data from <a href="http://fileserve.mrcieu.ac.uk/ld/1kg.v3.tgz">here</a>. That file will need to be unzipped.</p>
<p>An alternative is to use a built in function in the CAUSE R pacakge and precomputed pairwise estimates of <span class="math inline">\(r^2\)</span>. This is the method we used in our paper, coupled with LD estimates 1000 Genomes European samples computed via LDshrink. This method is slow but lets you use any LD estimates you want. We show how to do this in a section at the end of this document.</p>
<p>In either case, we prioritize variants based on their trait 1 p-value. We can limit ourselves to SNPs with trait 1 p-value less than 0.001 since we will use that threshold for estimating the CAUSE posteriors in the next step. If you use a higher threshold in the next step, you should also use a higher threshold in the pruning step. It is ok to have a lower threshold in the posterior estimation step than in the pruning step. In this case, the original GWAS data for LDL contains p-values se we use these. However, if these are missing you can compute approximate p-values using a normal approximation.</p>
<div id="ld-pruning-using-plink" class="section level3">
<h3>LD pruning using Plink</h3>
<p>The following code performs LD pruning using the <code>ieugwasr</code> wrapper to Plink LD clumping function. If you have Plink already installed, you can replace the path in <code>plink_bin</code> with a path to your local installation. To use the code below, you will need the <code>genetics.binaRies</code> package which can be installed with <code>devtools::install_github("explodecomputer/genetics.binaRies")</code>. If you want to use local reference data, you can set the <code>bfile</code> option of <code>ld_clump</code> (see <code>?ld_clump</code>).</p>
<pre class="r"><code>r2_thresh = 0.01
pval_thresh = 1e-3

X_clump &lt;- X %&gt;%
           rename(rsid = snp,
                  pval = p1) %&gt;%
           ieugwasr::ld_clump(dat = .,
                     clump_r2 = r2_thresh,
                     clump_p = pval_thresh,
                     plink_bin = genetics.binaRies::get_plink_binary(), 
                     pop = &quot;EUR&quot;)</code></pre>
<pre><code>API: public: http://gwas-api.mrcieu.ac.uk/</code></pre>
<pre><code>Please look at vignettes for options on running this locally if you need to run many instances of this command.</code></pre>
<pre><code>Clumping Zieg5o, 2023354 variants, using EUR population reference</code></pre>
<pre><code>Server code: 502; Server is possibly experiencing traffic, trying again...</code></pre>
<pre><code>Retry succeeded!</code></pre>
<pre><code>Removing 2022635 of 2023354 variants due to LD with other variants or absence from LD reference panel</code></pre>
<pre class="r"><code>top_vars &lt;- X_clump$rsid</code></pre>
<p>Here <code>r2_thresh</code> is the <span class="math inline">\(r^2\)</span> threshold you would like to use and <code>pval_thresh</code> is the p-value threshold you will use to compute the CAUSE posteriors. We recommend using a lower <span class="math inline">\(r^2\)</span> threshold for plink LD clumping than for LD clumping based on estimates from LDshrink, since these are not shrunken.</p>
</div>
</div>
<div id="step-4-fit-cause" class="section level2">
<h2>Step 4: Fit CAUSE</h2>
<p>Now that we have formatted data, an LD pruned set of variants, and nuisance parameters estimated, we can fit CAUSE. The function <code>cause</code> estimates posterior distributions under the sharing and causal models and calculates the ELPD for both models as well as for the null model in which there is neither a causal or a shared factor effect. This might take 5-10 minutes.</p>
<pre class="r"><code>res &lt;- cause(X=X, variants = top_vars, param_ests = params)</code></pre>
<pre><code>Estimating CAUSE posteriors using  719  variants.
Fitting confounder only model.
Setting Ranges
Refining Grid
Fitting causal model.
Setting Ranges
Refining Grid</code></pre>
<div id="pareto-k-diagnostics-warning" class="section level3">
<h3>Pareto k diagnostics warning</h3>
<p>Occasionally we see a warining about Pareto k diagnostics. This comes from the estimate of the elpd from the <code>loo</code> package. Usually we do not worry about it if there are few problematic samples for more details see help(‘pareto-k-diagnostic’). The <code>loo</code> objects are stored in a three element list <code>res$loos</code>. The first element is empty, the second element corresponds to the sharing model and the third element corresponds to the causal model. To print the Pareto k tables for the CAUSE models use</p>
<pre class="r"><code>res$loos[[2]]</code></pre>
<pre><code>
Computed from 1000 by 719 log-likelihood matrix

         Estimate    SE
elpd_loo    990.5  59.2
p_loo         0.7   0.1
looic     -1981.0 118.5
------
Monte Carlo SE of elpd_loo is 0.0.

All Pareto k estimates are good (k &lt; 0.5).
See help(&#39;pareto-k-diagnostic&#39;) for details.</code></pre>
<pre class="r"><code>loo::pareto_k_table(res$loos[[2]])</code></pre>
<pre><code>
All Pareto k estimates are good (k &lt; 0.5).</code></pre>
<pre class="r"><code>res$loos[[3]]</code></pre>
<pre><code>
Computed from 1000 by 719 log-likelihood matrix

         Estimate    SE
elpd_loo    997.6  58.9
p_loo         0.6   0.0
looic     -1995.2 117.8
------
Monte Carlo SE of elpd_loo is 0.0.

All Pareto k estimates are good (k &lt; 0.5).
See help(&#39;pareto-k-diagnostic&#39;) for details.</code></pre>
<pre class="r"><code>loo::pareto_k_table(res$loos[[3]])</code></pre>
<pre><code>
All Pareto k estimates are good (k &lt; 0.5).</code></pre>
<p>In some cases the problem can be resolved by fitting with more variants. If you are fitting with fewer than 1000 variants you could consider raising the p-value or <span class="math inline">\(r^2\)</span> thresholds and verifying that both studies have genome-wide coverage.</p>
</div>
</div>
<div id="step-5-look-at-results" class="section level2">
<h2>Step 5: Look at Results</h2>
<p>The resulting <code>cause</code> object contains an object for the partial sharing model fit (<code>sharing</code>), and object for the causal model fit (<code>causal</code>) and a table of ELPD results.</p>
<pre class="r"><code>class(res)</code></pre>
<pre><code>[1] &quot;cause&quot;</code></pre>
<pre class="r"><code>names(res)</code></pre>
<pre><code>[1] &quot;sharing&quot; &quot;causal&quot;  &quot;elpd&quot;    &quot;loos&quot;    &quot;data&quot;    &quot;sigma_g&quot; &quot;qalpha&quot; 
[8] &quot;qbeta&quot;  </code></pre>
<pre class="r"><code>res$elpd</code></pre>
<pre><code>   model1  model2 delta_elpd se_delta_elpd         z
1    null sharing -29.660651      5.588301 -5.307633
2    null  causal -36.735730      7.092086 -5.179821
3 sharing  causal  -7.075079      1.631256 -4.337198</code></pre>
<pre class="r"><code>class(res$sharing)</code></pre>
<pre><code>[1] &quot;cause_post&quot;</code></pre>
<pre class="r"><code>class(res$causal)</code></pre>
<pre><code>[1] &quot;cause_post&quot;</code></pre>
<p>The <code>elpd</code> table has the following columns:</p>
<ul>
<li>model1, model2: The models being compared</li>
<li>delta_elpd: Estimated difference in elpd. If delta_elpd is negative, model 2 is a better fit</li>
<li>se_delta_elpd: Estimated standard error of delta_elpd</li>
<li>z: delta_elpd/se_delta_elpd. A z-score that can be compared to a normal distribution to test if the difference in model fit is significant.</li>
</ul>
<p>In this case we see that the causal model is significantly better than the sharing model from the thrid line of the table. The <span class="math inline">\(z\)</span>-score is -4.34 corresponding to a p-value of 7.2^{-6}.</p>
<p>For each model (partial sharing and full) we can plot the posterior distributions of the parameters. Dotted lines show the prior distributions.</p>
<pre class="r"><code>plot(res$sharing)</code></pre>
<p><img src="figure/ldl_cad.Rmd/plot1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot1-1">
Past versions of plot1-1.png
</button>
</p>
<div id="fig-plot1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/05b1882c3b4bb8ae03f0eab05b6ec880750a1247/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">05b1882</a>
</td>
<td>
Jean Morrison
</td>
<td>
2021-07-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/81a1a48e3401219d0fea62242133be7117dc8656/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">81a1a48</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/70e2b97cc2ac9ecda4f2c3bc9dd1aaae04e00276/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">70e2b97</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-12-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/bfa7d2888b1ddb77a0b565246f2ba6b7c0a7d007/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">bfa7d28</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/d8d148608a68c9fd853964a999d3ac3e216978cc/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">d8d1486</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/33b373230107121d24eb71d1d4a7bdd70488f4c4/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">33b3732</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/286f4e9e0a432565b96c38bba3a1b155c02f9a78/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">286f4e9</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/1753b227154a949d80ae0ece3d13ef18f92f29b7/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">1753b22</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/2eb09d8a6a0e5cb986f1ffd6a71a6e44ca2567b9/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">2eb09d8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/4a8f76ccd2554d1f46985507b41899ced9e02a3e/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">4a8f76c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/a34393dedf81051e37783951c9e3b74aba80de65/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">a34393d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/bbe49018f1c79ea1f2bfb1c1c4ac2228722dac60/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">bbe4901</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/73690eb831172fcd7130e9b4d2620f85b0c528c1/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">73690eb</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot(res$causal)</code></pre>
<p><img src="figure/ldl_cad.Rmd/plot1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot1-2">
Past versions of plot1-2.png
</button>
</p>
<div id="fig-plot1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/05b1882c3b4bb8ae03f0eab05b6ec880750a1247/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">05b1882</a>
</td>
<td>
Jean Morrison
</td>
<td>
2021-07-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/81a1a48e3401219d0fea62242133be7117dc8656/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">81a1a48</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/70e2b97cc2ac9ecda4f2c3bc9dd1aaae04e00276/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">70e2b97</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-12-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/bfa7d2888b1ddb77a0b565246f2ba6b7c0a7d007/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">bfa7d28</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/d8d148608a68c9fd853964a999d3ac3e216978cc/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">d8d1486</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/33b373230107121d24eb71d1d4a7bdd70488f4c4/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">33b3732</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/286f4e9e0a432565b96c38bba3a1b155c02f9a78/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">286f4e9</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/1753b227154a949d80ae0ece3d13ef18f92f29b7/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">1753b22</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/2eb09d8a6a0e5cb986f1ffd6a71a6e44ca2567b9/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">2eb09d8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/4a8f76ccd2554d1f46985507b41899ced9e02a3e/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">4a8f76c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/a34393dedf81051e37783951c9e3b74aba80de65/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">a34393d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/bbe49018f1c79ea1f2bfb1c1c4ac2228722dac60/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">bbe4901</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/73690eb831172fcd7130e9b4d2620f85b0c528c1/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">73690eb</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The <code>summary</code> method will summarize the posterior medians and credible intervals.</p>
<pre class="r"><code>summary(res, ci_size=0.95)</code></pre>
<pre><code>p-value testing that causal model is a better fit:  7.2e-06 
Posterior medians and  95 % credible intervals:
     model     gamma              eta                   q                  
[1,] &quot;Sharing&quot; NA                 &quot;0.39 (0.3, 0.51)&quot;    &quot;0.62 (0.43, 0.79)&quot;
[2,] &quot;Causal&quot;  &quot;0.32 (0.24, 0.4)&quot; &quot;-0.03 (-1.28, 0.91)&quot; &quot;0.03 (0, 0.24)&quot;   </code></pre>
<p>The <code>plot</code> method applied to a <code>cause</code> object will arrange all of this information on one spread.</p>
<pre class="r"><code>plot(res)</code></pre>
<p><img src="figure/ldl_cad.Rmd/plot2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot2-1">
Past versions of plot2-1.png
</button>
</p>
<div id="fig-plot2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/05b1882c3b4bb8ae03f0eab05b6ec880750a1247/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">05b1882</a>
</td>
<td>
Jean Morrison
</td>
<td>
2021-07-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/81a1a48e3401219d0fea62242133be7117dc8656/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">81a1a48</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/70e2b97cc2ac9ecda4f2c3bc9dd1aaae04e00276/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">70e2b97</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-12-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/1a90dd649497a3c846f8abc2832de648ac881e80/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">1a90dd6</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/bfa7d2888b1ddb77a0b565246f2ba6b7c0a7d007/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">bfa7d28</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/d8d148608a68c9fd853964a999d3ac3e216978cc/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">d8d1486</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/33b373230107121d24eb71d1d4a7bdd70488f4c4/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">33b3732</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/286f4e9e0a432565b96c38bba3a1b155c02f9a78/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">286f4e9</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/1753b227154a949d80ae0ece3d13ef18f92f29b7/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">1753b22</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/2eb09d8a6a0e5cb986f1ffd6a71a6e44ca2567b9/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">2eb09d8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/4a8f76ccd2554d1f46985507b41899ced9e02a3e/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">4a8f76c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/a34393dedf81051e37783951c9e3b74aba80de65/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">a34393d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/bbe49018f1c79ea1f2bfb1c1c4ac2228722dac60/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">bbe4901</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/73690eb831172fcd7130e9b4d2620f85b0c528c1/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">73690eb</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The <code>plot</code> method can also produce scatter plots of the data showing for each model, the probability that each variant is acting through the shared factor and the contribution of each variant to the ELPD test statistic.</p>
<pre class="r"><code>plot(res, type=&quot;data&quot;)</code></pre>
<p><img src="figure/ldl_cad.Rmd/plot3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot3-1">
Past versions of plot3-1.png
</button>
</p>
<div id="fig-plot3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/05b1882c3b4bb8ae03f0eab05b6ec880750a1247/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">05b1882</a>
</td>
<td>
Jean Morrison
</td>
<td>
2021-07-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/da5b784adce5647fa0d8d5ef8dc7dd8ecc6a92a4/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">da5b784</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/81a1a48e3401219d0fea62242133be7117dc8656/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">81a1a48</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/1cfcb0b3cc513064b0fecc385bb2317e482c6718/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">1cfcb0b</a>
</td>
<td>
Jean Morrison
</td>
<td>
2020-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/70e2b97cc2ac9ecda4f2c3bc9dd1aaae04e00276/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">70e2b97</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-12-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/1a90dd649497a3c846f8abc2832de648ac881e80/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">1a90dd6</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/bfa7d2888b1ddb77a0b565246f2ba6b7c0a7d007/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">bfa7d28</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/d8d148608a68c9fd853964a999d3ac3e216978cc/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">d8d1486</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/33b373230107121d24eb71d1d4a7bdd70488f4c4/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">33b3732</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/286f4e9e0a432565b96c38bba3a1b155c02f9a78/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">286f4e9</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-06-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/1cc48608917d72a60217a93bfb658e467af6c3ed/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">1cc4860</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/2eb09d8a6a0e5cb986f1ffd6a71a6e44ca2567b9/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">2eb09d8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/4a8f76ccd2554d1f46985507b41899ced9e02a3e/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">4a8f76c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ld-pruning-using-built-in-function" class="section level2">
<h2>LD pruning using built in function</h2>
<p>The function <code>ld_prune</code> uses a greedy algorithm that selects the variant with the lowest LDL p-value and removes all variants in LD with the selected variant and then repeats until no variants are left. This step requires LD estimates. You can download estimates made in the 1000 Genomes CEU population <a href="https://zenodo.org/record/1464357#.W8a-fxROmV4">here</a>. We first demonstrate use of the function for one chromosome and then show an example of how to parallelize the analysis over all 22 autosomes.</p>
<pre class="r"><code>download.file(&quot;https://zenodo.org/record/1464357/files/chr22_AF0.05_0.1.RDS?download=1&quot;, destfile = &quot;example_data/chr22_AF0.05_0.1.RDS&quot;)
download.file(&quot;https://zenodo.org/record/1464357/files/chr22_AF0.05_snpdata.RDS?download=1&quot;, destfile=&quot;example_data/chr22_AF0.05_snpdata.RDS&quot;)</code></pre>
<pre class="r"><code>ld &lt;- readRDS(&quot;example_data/chr22_AF0.05_0.1.RDS&quot;)
snp_info &lt;- readRDS(&quot;example_data/chr22_AF0.05_snpdata.RDS&quot;)

head(ld)</code></pre>
<pre><code>      rowsnp      colsnp        r2
1 rs62224609 rs376238049 0.9012642
2 rs62224609  rs62224614 0.9907366
3 rs62224609   rs7286962 0.9907366
4 rs62224609  rs55926024 0.1103000
5 rs62224609 rs117246541 0.9012642
6 rs62224609  rs62224618 0.9907366</code></pre>
<pre class="r"><code>head(snp_info)</code></pre>
<pre><code># A tibble: 6 × 9
     AF SNP         allele    chr      pos   snp_id region_id   map ld_snp_id
  &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;   &lt;int&gt;    &lt;int&gt;    &lt;int&gt;     &lt;int&gt; &lt;dbl&gt;     &lt;int&gt;
1 0.884 rs62224609  T,C        22 16051249 79758556        22     0  79758556
2 0.904 rs4965031   G,A        22 16052080 79758578        22     0  79758578
3 0.646 rs375684679 A,AAAAC    22 16052167 79758584        22     0  79758584
4 0.894 rs376238049 C,T        22 16052962 79758602        22     0  79758602
5 0.934 rs200777521 C,A        22 16052986 79758604        22     0  79758604
6 0.934 rs80167676  A,T        22 16053444 79758627        22     0  79758627</code></pre>
<p>The <code>ld</code> data frame should contain the column names <code>rowsnp</code>, <code>colsnp</code>, and <code>r2</code>. The <code>snp_info</code> data frame contains information about all of the chromosome 22 variants with allele frequency greater than 0.05. The only piece of information we need from this data frame is the list of variants <code>snp_info$SNP</code> which provides the total list of variants used in the LD calculations.</p>
<div id="ld-pruning-for-one-chromosome" class="section level3">
<h3>LD pruning for one chromosome</h3>
<p>The <code>ld_prune</code> function is somewhat flexible in its arguments, see <code>help(ld_prune)</code>.</p>
<pre class="r"><code>variants &lt;- X %&gt;% mutate(pval1 = 2*pnorm(abs(beta_hat_1/seb1), lower.tail=FALSE))
pruned &lt;- ld_prune(variants = variants, 
                            ld = ld, total_ld_variants = snp_info$SNP, 
                            pval_cols = c(&quot;pval1&quot;), 
                            pval_thresh = c(1e-3))
length(pruned)</code></pre>
<p>If <code>length(pval_cols) =1</code>, <code>ld_prune</code> returns a vector of selected variants. If There are multipld <span class="math inline">\(p\)</span>-value columns provided, <code>ld_prune</code> will return a list of vectors, one for each column. The length of <code>pval_thresh</code> should be equal to the length of <code>pval_cols</code> and provides a threshold for each column. Excluding variants with high <span class="math inline">\(p\)</span>-values speeds up the pruning step. We can fit CAUSE without high <span class="math inline">\(p\)</span>-value variants because these variants have almost no influence on the posterior distributions or the ELPD test statistic. The exact value of the threshold isn’t important as long as it is fairly lenient. Including additional variants may slow down computation but shouldn’t change the results</p>
</div>
<div id="parallelizing-over-chromosomes" class="section level3">
<h3>Parallelizing over chromosomes</h3>
<p>We highly recommend parallelizing for whole genome LD pruning. One way to do this is with the parallel pacakge, though this option uses a lot of memory. A better option is to submit separate jobs to nodes of a compute cluster and then combine results.</p>
<p>If you are analyzing many phenotypes, the most efficient way to complete this step is to first obtain a list of variants present in all data sets and use only these variants in your analysis. You can then obtain an LD-pruned set of variants prioritized for low <span class="math inline">\(p\)</span>-values for each traits (<span class="math inline">\(N\)</span> lists if there are <span class="math inline">\(N\)</span> phenotypes). In any analysis, you will use the list of variants prioritized for the trait <span class="math inline">\(M\)</span> phenotype.</p>
<p>Download the LD-pruned variant list for our example analysis: <a href="https://github.com/jean997/cause/blob/master/example_data/top_ldl_pruned_vars.RDS">top LDL list</a></p>
<pre class="r"><code>download.file(&quot;https://github.com/jean997/cause/blob/master/example_data/top_ldl_pruned_vars.RDS&quot;, destfile = &quot;example_data/top_ldl_pruned_vars.RDS&quot;)</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] cause_1.2.0.0327 dplyr_1.0.7      readr_2.0.2      workflowr_1.6.2 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7         invgamma_1.1       lattice_0.20-44    tidyr_1.1.3       
 [5] assertthat_0.2.1   rprojroot_2.0.2    digest_0.6.28      utf8_1.2.2        
 [9] truncnorm_1.0-8    R6_2.5.1           evaluate_0.14      highr_0.9         
[13] httr_1.4.2         ggplot2_3.3.5      pillar_1.6.4       rlang_0.4.12      
[17] curl_4.3.2         rstudioapi_0.13    irlba_2.3.3        whisker_0.4       
[21] Matrix_1.3-4       rmarkdown_2.9      labeling_0.4.2     stringr_1.4.0     
[25] loo_2.4.1          bit_4.0.4          munsell_0.5.0      mixsqp_0.3-43     
[29] compiler_4.1.1     httpuv_1.6.1       xfun_0.24          pkgconfig_2.0.3   
[33] SQUAREM_2021.1     ieugwasr_0.1.5     htmltools_0.5.2    tidyselect_1.1.1  
[37] tibble_3.1.5       gridExtra_2.3      intervals_0.15.2   matrixStats_0.60.0
[41] fansi_0.5.0        crayon_1.4.1       tzdb_0.1.2         later_1.3.0       
[45] grid_4.1.1         jsonlite_1.7.2     gtable_0.3.0       lifecycle_1.0.1   
[49] DBI_1.1.1          git2r_0.28.0       magrittr_2.0.1     scales_1.1.1      
[53] RcppParallel_5.1.4 cli_3.0.1          stringi_1.7.5      vroom_1.5.5       
[57] farver_2.1.0       fs_1.5.0           promises_1.2.0.1   ellipsis_0.3.2    
[61] generics_0.1.0     vctrs_0.3.8        tools_4.1.1        bit64_4.0.5       
[65] glue_1.4.2         purrr_0.3.4        hms_1.1.1          parallel_4.1.1    
[69] fastmap_1.1.0      yaml_2.2.1         colorspace_2.0-2   ashr_2.2-47       
[73] knitr_1.33        </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] cause_1.2.0.0327 dplyr_1.0.7      readr_2.0.2      workflowr_1.6.2 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7         invgamma_1.1       lattice_0.20-44    tidyr_1.1.3       
 [5] assertthat_0.2.1   rprojroot_2.0.2    digest_0.6.28      utf8_1.2.2        
 [9] truncnorm_1.0-8    R6_2.5.1           evaluate_0.14      highr_0.9         
[13] httr_1.4.2         ggplot2_3.3.5      pillar_1.6.4       rlang_0.4.12      
[17] curl_4.3.2         rstudioapi_0.13    irlba_2.3.3        whisker_0.4       
[21] Matrix_1.3-4       rmarkdown_2.9      labeling_0.4.2     stringr_1.4.0     
[25] loo_2.4.1          bit_4.0.4          munsell_0.5.0      mixsqp_0.3-43     
[29] compiler_4.1.1     httpuv_1.6.1       xfun_0.24          pkgconfig_2.0.3   
[33] SQUAREM_2021.1     ieugwasr_0.1.5     htmltools_0.5.2    tidyselect_1.1.1  
[37] tibble_3.1.5       gridExtra_2.3      intervals_0.15.2   matrixStats_0.60.0
[41] fansi_0.5.0        crayon_1.4.1       tzdb_0.1.2         later_1.3.0       
[45] grid_4.1.1         jsonlite_1.7.2     gtable_0.3.0       lifecycle_1.0.1   
[49] DBI_1.1.1          git2r_0.28.0       magrittr_2.0.1     scales_1.1.1      
[53] RcppParallel_5.1.4 cli_3.0.1          stringi_1.7.5      vroom_1.5.5       
[57] farver_2.1.0       fs_1.5.0           promises_1.2.0.1   ellipsis_0.3.2    
[61] generics_0.1.0     vctrs_0.3.8        tools_4.1.1        bit64_4.0.5       
[65] glue_1.4.2         purrr_0.3.4        hms_1.1.1          parallel_4.1.1    
[69] fastmap_1.1.0      yaml_2.2.1         colorspace_2.0-2   ashr_2.2-47       
[73] knitr_1.33        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
