<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jean Morrison" />

<meta name="date" content="2018-10-16" />

<title>Example Analysis with CAUSE: LDL -&gt; CAD</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CAUSE: Cauasl Analysis Using Summary Effects</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="ldl_cad.html">R Package Tutorial</a>
</li>
<li>
  <a href="simulations.html">Simulations</a>
</li>
<li>
  <a href="gwas_pairs.html">Analyze Pairs of GWAS Traits</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jean997/cause">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Example Analysis with CAUSE: LDL -&gt; CAD</h1>
<h4 class="author">Jean Morrison</h4>
<h4 class="date">2018-10-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-07-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>cause/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0.9000). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongobjectspresent"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Environment:</strong> objects present </a>
</p>
</div>
<div id="strongEnvironmentstrongobjectspresent" class="panel-collapse collapse">
<div class="panel-body">
<p>
The global environment had objects present when the code in the R Markdown file was run. These objects can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment. Use <code>wflow_publish</code> or <code>wflow_build</code> to ensure that the code is always run in an empty environment.
</p>
<p>
The following objects were defined in the global environment when these results were created:
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
Name
</th>
<th style="text-align:left;">
Class
</th>
<th style="text-align:left;">
Size
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
data
</td>
<td style="text-align:left;">
environment
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
env
</td>
<td style="text-align:left;">
environment
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20181014code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20181014)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20181014code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20181014)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
look
</li>
<li>
params
</li>
<li>
plot1
</li>
<li>
plot2
</li>
<li>
readdata
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
summary
</li>
<li>
unnamed-chunk-10
</li>
<li>
unnamed-chunk-11
</li>
<li>
unnamed-chunk-12
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-5
</li>
<li>
unnamed-chunk-6
</li>
<li>
unnamed-chunk-7
</li>
<li>
unnamed-chunk-8
</li>
<li>
unnamed-chunk-9
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>ldl_cad_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjean997causetreed8f89fc6d0b98cfb193685a5aa3eba2f92d5a3a2targetblankd8f89fca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jean997/cause/tree/d8f89fc6d0b98cfb193685a5aa3eba2f92d5a3a2" target="_blank">d8f89fc</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjean997causetreed8f89fc6d0b98cfb193685a5aa3eba2f92d5a3a2targetblankd8f89fca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:   (tumble-track's conflicted copy 2019-07-09).Rhistory
    Untracked:  DESCRIPTION (tumble-track's conflicted copy 2019-07-09)
    Untracked:  analysis/gwas_pairs.Rmd
    Untracked:  analysis/ldl_cad_cache/
    Untracked:  analysis/simulations.Rmd
    Untracked:  docs/WNAR_2019.pdf
    Untracked:  example_data/chr22_AF0.05_0.1.RDS
    Untracked:  example_data/chr22_AF0.05_snpdata.RDS
    Untracked:  gwas_data/
    Untracked:  ll_v7_notes.Rmd
    Untracked:  src/RcppExports.o
    Untracked:  src/log_likelihood_functions.o
    Untracked:  temp.txt

Unstaged changes:
    Modified:   R/map_pi_rho.R
    Modified:   analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd
    Deleted:    test_mixSQP.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/d8f89fc6d0b98cfb193685a5aa3eba2f92d5a3a2/analysis/ldl_cad.Rmd" target="_blank">d8f89fc</a>
</td>
<td>
Jean Morrison
</td>
<td>
2019-07-10
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/1753b227154a949d80ae0ece3d13ef18f92f29b7/docs/ldl_cad.html" target="_blank">1753b22</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-10
</td>
<td>
update website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/2eb09d8a6a0e5cb986f1ffd6a71a6e44ca2567b9/docs/ldl_cad.html" target="_blank">2eb09d8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
<td>
added new option for a truncated prior on q
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/4a8f76ccd2554d1f46985507b41899ced9e02a3e/docs/ldl_cad.html" target="_blank">4a8f76c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/2652be12bb63e67611ef6eee0647254b790ba52a/analysis/ldl_cad.Rmd" target="_blank">2652be1</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/6ae7a60c30167592b1328cc1ee64958f29086ee1/docs/ldl_cad.html" target="_blank">6ae7a60</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-25
</td>
<td>
build website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/48bdf215b8c526c53cc9a0308677675ef231b2e9/analysis/ldl_cad.Rmd" target="_blank">48bdf21</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-25
</td>
<td>
fixing warnings
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/8c57c8a3cc8b0faa204a5dfa1a89336039622947/analysis/ldl_cad.Rmd" target="_blank">8c57c8a</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-25
</td>
<td>
build website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/8c57c8a3cc8b0faa204a5dfa1a89336039622947/docs/ldl_cad.html" target="_blank">8c57c8a</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-25
</td>
<td>
build website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/a34393dedf81051e37783951c9e3b74aba80de65/analysis/ldl_cad.Rmd" target="_blank">a34393d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-25
</td>
<td>
build website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/a34393dedf81051e37783951c9e3b74aba80de65/docs/ldl_cad.html" target="_blank">a34393d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-25
</td>
<td>
build website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/6354c3568e965503bbdde96d84c04c8801405bfd/docs/ldl_cad.html" target="_blank">6354c35</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/24510f128068f566d4efca738364151712ceda03/analysis/ldl_cad.Rmd" target="_blank">24510f1</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-23
</td>
<td>
small language changes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/d050da86b8956c154f24ef61a9c1ad270b7702b2/analysis/ldl_cad.Rmd" target="_blank">d050da8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-23
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/bbe49018f1c79ea1f2bfb1c1c4ac2228722dac60/docs/ldl_cad.html" target="_blank">bbe4901</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/558cd32e08be455b3a9cca07c0caf5515b3a8ca2/analysis/ldl_cad.Rmd" target="_blank">558cd32</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jean997/cause/73690eb831172fcd7130e9b4d2620f85b0c528c1/docs/ldl_cad.html" target="_blank">73690eb</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/1a891e3b8f5e85ae882d7b6b31fb0f590825c34c/analysis/ldl_cad.Rmd" target="_blank">1a891e3</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
wflow_publish(“analysis/ldl_cad.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jean997/cause/blob/d10191f6c1e324ef6073a95893546ccd9474bed1/analysis/ldl_cad.Rmd" target="_blank">d10191f</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
<td>
wflow_git_commit(“analysis/ldl_cad.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This document will walk through a real genome-sized example of how to use CAUSE. Some of the steps will take 5-10 minutes. The LD pruning steps will benefit from access to a cluster or multiple cores. For steps that require long computation we also provide output files that can be downloaded to make it easier to run through the example.</p>
<p>We will be analyzing GWAS data for LDL cholesterol and for coronary artery disease to test for a causal relationship of LDL on CAD. The analysis will have the following steps:</p>
<ol style="list-style-type: decimal">
<li>Format the data for use with CAUSE</li>
<li>Calculate nuisance parameters</li>
<li>LD pruning</li>
<li>Fit CAUSE</li>
<li>Look at results</li>
</ol>
<p>Step 3 will require LD information estimated from the 1000 Genomes CEU population using LDshrink <a href="https://zenodo.org/record/1464357#.W8a-fxROmV4">here</a>. LD data are about 11 Gb. The GWAS data we will use are about 320 Gb. However, in this tutorial you will be able to skip the large data steps and simply download the results.</p>
</div>
<div id="step-0-install-cause" class="section level2">
<h2>Step 0: Install CAUSE</h2>
<p>In R</p>
<pre class="r"><code>devtools::install_github(&quot;jean997/cause&quot;)</code></pre>
<p>Please be sure you are using <code>mixsqp-0-97</code> which is currently the version available in CRAN *not the latest version available on GitHub.</p>
</div>
<div id="step-1-format-data-for-cause" class="section level2">
<h2>Step 1: Format Data for CAUSE</h2>
<p>We will use <code>read_tsv</code> to read in summary statistics for a GWAS of LDL cholesterol and a GWAS of coronary artery disease from the internet. We will then combine these and format them for use with CAUSE. First read in the data. For LDL Cholesterol, we use summary statistics from Willer et al (2013) [PMID: 24097068]. For CAD we use summary statistics from van der Harst et al. (2017) [PMID: 29212778]</p>
<pre class="r"><code>library(readr)
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(cause)</code></pre>
<pre class="r"><code>X1 &lt;- read_tsv(&quot;http://csg.sph.umich.edu/abecasis/public/lipids2013/jointGwasMc_LDL.txt.gz&quot;)
X2 &lt;- read_tsv(&quot;ftp://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/vanderHarstP_29212778_GCST005194/CAD_META.gz&quot;)</code></pre>
<p>CAUSE needs the following information from each data set: SNP or variant ID, effect size, and standard error, effect allele and other allele. For convenience, we provide a simple function that will merge data sets and produce a <code>cause_data</code> object that can be used with later functions. This step and the rest of the analysis are done in R.</p>
<p>The function <code>gwas_format_cause</code> will try to merge two data sets and and align effect sizes to correspond to the same allele. It will remove variants with ambiguous alleles (G/C or A/T) or with alleles that do not match between data sets (e.g A/G in one data set and A/C in the other). It will not remove variants that are simply strand flipped between the two data sets (e. g. A/C in one data set, T/G in the other).</p>
<p>LDL column headers:</p>
<ul>
<li>SNP: rsid</li>
<li>Effect: beta</li>
<li>Standard Error: se</li>
<li>Effect Allele: A1</li>
<li>Other Allele: A2</li>
</ul>
<p>CAD column headers:</p>
<ul>
<li>SNP: oldID</li>
<li>Effect: Effect</li>
<li>Standard Error: StdErr</li>
<li>Effect Allele: Allele1</li>
<li>Other Allele: Allele2</li>
</ul>
<pre class="r"><code>X &lt;- gwas_format_cause(X1, X2, snp_name_cols = c(&quot;rsid&quot;, &quot;oldID&quot;), 
                       beta_hat_cols = c(&quot;beta&quot;, &quot;Effect&quot;), 
                       se_cols = c(&quot;se&quot;, &quot;StdErr&quot;), 
                       A1_cols = c(&quot;A1&quot;, &quot;Allele1&quot;), 
                       A2_cols = c(&quot;A2&quot;, &quot;Allele2&quot;))</code></pre>
<p>Alternatively, you can download already formatted data <a href="https://github.com/jean997/cause/blob/master/example_data/LDL_CAD_merged.RDS">here</a> and read it in using <code>readRDS</code>.</p>
<pre class="r"><code>system(&quot;mkdir example_data/&quot;)
download.file(&quot;https://github.com/jean997/cause/blob/master/example_data/LDL_CAD_merged.RDS&quot;, destfile = &quot;example_data/LDL_CAD_merged.RDS&quot;)</code></pre>
<pre class="r"><code>X &lt;- readRDS(&quot;example_data/LDL_CAD_merged.RDS&quot;)</code></pre>
<pre class="r"><code>head(X)</code></pre>
<pre><code>        snp beta_hat_1   seb1 beta_hat_2   seb2 A1 A2
1 rs4747841     0.0037 0.0052     0.0106 0.0056  a  G
2 rs4749917    -0.0033 0.0052    -0.0108 0.0056  a  G
3  rs737656     0.0099 0.0054     0.0196 0.0058  a  G
4  rs737657     0.0084 0.0054     0.0195 0.0058  a  G
5 rs7086391    -0.0075 0.0067     0.0115 0.0072  a  G
6  rs878177    -0.0073 0.0055    -0.0225 0.0059  a  G</code></pre>
<p>There are likely more efficient ways to do this merge. If you would like to process the data yourself, you can construct a <code>cause_data</code> object from a data frame using the constructor <code>new_cause_data(X)</code> where <code>X</code> is any data frame that includes the columns <code>snp</code>, <code>beta_hat_1</code>, <code>seb1</code>, <code>beta_hat_2</code>, and <code>seb2</code>.</p>
</div>
<div id="step-2-calculate-nuisance-parameters" class="section level2">
<h2>Step 2: Calculate nuisance parameters</h2>
<p>The next step is to estimate the parameters that define the prior distribution of <span class="math inline">\(\beta_{M}\)</span> and <span class="math inline">\(\theta\)</span> and to estimate <span class="math inline">\(\rho\)</span>, the correlation between summary statistics that is due to sample overlap or population structure. We will do this with a random subset of 1,000,000 variants since our data set is large. <code>est_cause_params</code> estimates the nuisance parameters by finding the maximum a posteriori estimate of <span class="math inline">\(\rho\)</span> and the mixing parameters when <span class="math inline">\(\gamma = \eta = 0\)</span>. This step takes a several minutes.</p>
<pre class="r"><code>set.seed(100)
varlist &lt;- with(X, sample(snp, size=1000000, replace=FALSE))
params &lt;- est_cause_params(X, varlist)</code></pre>
<pre><code>Estimating CAUSE parameters with  1000000  variants.
1 0.1255649 
2 0.000503845 
3 6.082345e-06 
4 7.406906e-08 </code></pre>
<p>The object <code>params</code> is of class <code>cause_params</code> and contains information about the fit as well as the maximum a posteriori estimates of the mixing parameters (<span class="math inline">\(\pi\)</span>) and <span class="math inline">\(\rho\)</span>. The object <code>params$mix_grid</code> is a data frame defining the distribution of summary statistics. The column <code>S1</code> is the variance for trait 1 (<span class="math inline">\(M\)</span>), the column <code>S2</code> is the variance for trait 2 (<span class="math inline">\(Y\)</span>) and the column <code>pi</code> is the mixture proportion assigned to that variance combination.</p>
<pre class="r"><code>class(params)</code></pre>
<pre><code>[1] &quot;cause_params&quot;</code></pre>
<pre class="r"><code>names(params)</code></pre>
<pre><code> [1] &quot;rho&quot;       &quot;pi&quot;        &quot;mix_grid&quot;  &quot;loglik&quot;    &quot;PIS&quot;      
 [6] &quot;RHO&quot;       &quot;LLS&quot;       &quot;converged&quot; &quot;prior&quot;     &quot;var&quot;      </code></pre>
<pre class="r"><code>params$rho</code></pre>
<pre><code>[1] 0.06515036</code></pre>
<pre class="r"><code>head(params$mix_grid)</code></pre>
<pre><code>           S1          S2         pi
1 0.000000000 0.000000000 0.31994147
2 0.000000000 0.003419647 0.20495624
3 0.000000000 0.004836112 0.17805780
4 0.003533615 0.004836112 0.16065491
5 0.004997287 0.009672224 0.11635100
6 0.009994574 0.009672224 0.00188234</code></pre>
<p>So, for example, in this case, we have estimated that 32% of variants have trait 1 variance and trait 2 equal to 0 meaning that they have no association with either trait.</p>
<p>Tip: Do not try to estimate the nuisance parameters with substantially fewer than 100,000 variants. This can lead to poor estimates of the mixing parameters whih leads to bad model comparisons.</p>
</div>
<div id="step-3-ld-pruning" class="section level2">
<h2>Step 3: LD Pruning</h2>
<p>We estimate CAUSE posterior distributions using an LD pruned set of variants, prioritizing variants with low trait <span class="math inline">\(M\)</span> (LDL) <span class="math inline">\(p\)</span>-values. The CAUSE R package contains a function to help do this. The function <code>ld_prune</code> uses a greedy algorithm that selects the variant wtih the lowest LDL p-value and removes all variants in LD with the selected variant and then repeats until no variants are left. This step requires LD estimates. You can download estimates made in the 1000 Genomes CEU population <a href="https://zenodo.org/record/1464357#.W8a-fxROmV4">here</a>. We first demonstrate use of the function for one chromosome and then show an example of how to parallelize the analysis over all 22 autosomes.</p>
<pre class="r"><code>download.file(&quot;https://zenodo.org/record/1464357/files/chr22_AF0.05_0.1.RDS?download=1&quot;, destfile = &quot;example_data/chr22_AF0.05_0.1.RDS&quot;)
download.file(&quot;https://zenodo.org/record/1464357/files/chr22_AF0.05_snpdata.RDS?download=1&quot;, destfile=&quot;example_data/chr22_AF0.05_snpdata.RDS&quot;)</code></pre>
<pre class="r"><code>ld &lt;- readRDS(&quot;example_data/chr22_AF0.05_0.1.RDS&quot;)
snp_info &lt;- readRDS(&quot;example_data/chr22_AF0.05_snpdata.RDS&quot;)

head(ld)</code></pre>
<pre><code>      rowsnp      colsnp        r2
1 rs62224609 rs376238049 0.9012642
2 rs62224609  rs62224614 0.9907366
3 rs62224609   rs7286962 0.9907366
4 rs62224609  rs55926024 0.1103000
5 rs62224609 rs117246541 0.9012642
6 rs62224609  rs62224618 0.9907366</code></pre>
<pre class="r"><code>head(snp_info)</code></pre>
<pre><code># A tibble: 6 x 9
     AF SNP        allele    chr      pos  snp_id region_id   map ld_snp_id
  &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;   &lt;int&gt;    &lt;int&gt;   &lt;int&gt;     &lt;int&gt; &lt;dbl&gt;     &lt;int&gt;
1 0.884 rs62224609 T,C        22 16051249  7.98e7        22     0  79758556
2 0.904 rs4965031  G,A        22 16052080  7.98e7        22     0  79758578
3 0.646 rs3756846… A,AAAAC    22 16052167  7.98e7        22     0  79758584
4 0.894 rs3762380… C,T        22 16052962  7.98e7        22     0  79758602
5 0.934 rs2007775… C,A        22 16052986  7.98e7        22     0  79758604
6 0.934 rs80167676 A,T        22 16053444  7.98e7        22     0  79758627</code></pre>
<p>The <code>ld</code> data frame should contain the column names <code>rowsnp</code>, <code>colsnp</code>, and <code>r2</code>. The <code>snp_info</code> data frame contains information about all of the chromosome 22 variants with allele frequency greater than 0.05. The only piece of information we need from this data frame is the list of variants <code>snp_info$SNP</code> which provides the total list of variants used in the LD calculations.</p>
<div id="ld-pruning-for-one-chromosome" class="section level3">
<h3>LD pruning for one chromosome</h3>
<p>The <code>ld_prune</code> function is somewhat flexible in its arguments, see <code>help(ld_prune)</code>.</p>
<pre class="r"><code>variants &lt;- X %&gt;% mutate(pval1 = 2*pnorm(abs(beta_hat_1/seb1), lower.tail=FALSE))
pruned &lt;- ld_prune(variants = variants, 
                            ld = ld, total_ld_variants = snp_info$SNP, 
                            pval_cols = c(&quot;pval1&quot;), 
                            pval_thresh = c(1e-3))</code></pre>
<pre><code>You have suppplied information for  2023362  variants.
Of these,  22835  have LD information.</code></pre>
<pre class="r"><code>length(pruned)</code></pre>
<pre><code>[1] 15</code></pre>
<p><code>ld_prune</code> retunrs a list of vectors of length equal to the length of the <code>pval_cols</code> argument. In this case <code>pval_cols= c(NA, &quot;pval1&quot;)</code> meaning that the first element of <code>pruned</code> will be a randomly pruned list and the second will be pruned preferentially choosing variants with low values of <code>pval1</code>. We also apply a threshold specified by the <code>pval_thresh</code> argument. For the first list there is no threshold. For the second, the threshold is 0.001.</p>
</div>
<div id="parallelizing-over-chromosomes" class="section level3">
<h3>Parallelizing over chromosomes</h3>
<p>We highly recommend parallelizing for whole genome LD pruning. One way to do this is with the parallel pacakge, though this option uses a lot of memory.</p>
<pre class="r"><code>library(parallel)
cores &lt;- parallel::detectCores()-1
ld_files &lt;- paste0(&quot;example_data/ld_data/chr&quot;, 1:22, &quot;_AF0.05_0.1.RDS&quot;)
snp_info_files &lt;- paste0(&quot;example_data/ld_data/chr&quot;, 1:22, &quot;_AF0.05_snpdata.RDS&quot;)

cl &lt;- makeCluster(cores, type=&quot;PSOCK&quot;)
clusterExport(cl, varlist=c(&quot;variants&quot;, &quot;ld_files&quot;, &quot;snp_info_files&quot;))
clusterEvalQ(cl, library(cause))
system.time(
pruned &lt;- parLapply(cl, seq_along(ld_files[20:22]), fun=function(i){
  ld &lt;- readRDS(ld_files[i])
  snp_info &lt;- readRDS(snp_info_files[i])
  ld_prune(variants = variants, 
          ld = ld, total_ld_variants = snp_info$SNP, 
          pval_cols = c(&quot;pval1&quot;), 
          pval_thresh = c( 1e-3))
})
)
stopCluster(cl)

top_ldl_pruned_vars &lt;- unlist(pruned)</code></pre>
<p>A better option is to parallelize over the nodes of a compute cluster and then combine results.</p>
<p>Tip: If you are analyzing many phenotypes first obtain a list of variants present in all data sets and use only these variants in your analysis. This will allow you to do only <span class="math inline">\(N\)</span> LD pruning steps where <span class="math inline">\(N\)</span> is the number of phenotypes rather than <span class="math inline">\(N\)</span> choose 2.</p>
<p>Download the resulting variant list: <a href="https://github.com/jean997/cause/blob/master/example_data/top_ldl_pruned_vars.RDS">top LDL list</a></p>
<pre class="r"><code>download.file(&quot;https://github.com/jean997/cause/blob/master/example_data/top_ldl_pruned_vars.RDS&quot;, destfile = &quot;example_data/top_ldl_pruned_vars.RDS&quot;)</code></pre>
</div>
</div>
<div id="step-4-fit-cause" class="section level2">
<h2>Step 4: Fit CAUSE</h2>
<p>Now that we have formatted data, an LD pruned set of variants, and nuisance parameters estimated, we can fit CAUSE! The function <code>cause::cause</code> will estimate posterior distributions under the confounding and causal models and calculate the elpd for both models as well as for the null model in which there is neither a causal or a confounding effect. This might take 5-10 minutes.</p>
<p>Note: To estimate the posterior distributions, we only need the variants that are most associated with the mediator. This is because other variants don’t add any information about the relationship between the traits. When we LD pruned, we used a <span class="math inline">\(p\)</span>-value threshold of 0.001. The exact value of this threshold isn’t important as long as it is fairly lenient. Including additional variants may slow down computation but shouldn’t change the results</p>
<pre class="r"><code>top_vars &lt;- readRDS(&quot;example_data/top_ldl_pruned_vars.RDS&quot;)
res &lt;- cause(X=X, variants = top_vars, param_ests = params)</code></pre>
<pre><code>Estimating CAUSE posteriors using  1215  variants.</code></pre>
</div>
<div id="step-5-look-at-results" class="section level2">
<h2>Step 5: Look at Results</h2>
<p>The resulting <code>cause</code> object contains an object for the partial sharing model fit (<code>conf</code>), and object for the causal model fit (<code>full</code>) and a table of ELPD results.</p>
<pre class="r"><code>class(res)</code></pre>
<pre><code>[1] &quot;cause&quot;</code></pre>
<pre class="r"><code>names(res)</code></pre>
<pre><code>[1] &quot;conf&quot;    &quot;full&quot;    &quot;elpd&quot;    &quot;loos&quot;    &quot;data&quot;    &quot;sigma_g&quot; &quot;qalpha&quot; 
[8] &quot;qbeta&quot;  </code></pre>
<pre class="r"><code>res$elpd</code></pre>
<pre><code>  model1 model2 delta_elpd se_delta_elpd         z
1   null   conf -63.488710      7.978181 -7.957793
2   null   full -71.880159      9.175262 -7.834126
3   conf   full  -8.391449      1.285055 -6.530034</code></pre>
<pre class="r"><code>class(res$conf)</code></pre>
<pre><code>[1] &quot;cause_post&quot;</code></pre>
<pre class="r"><code>class(res$full)</code></pre>
<pre><code>[1] &quot;cause_post&quot;</code></pre>
<p>The <code>elpd</code> table has the following columns:</p>
<ul>
<li>model1, model2: The models being compared</li>
<li>delta_elpd: Estimated difference in elpd. If delta_elpd is negative, model 2 is a better fit</li>
<li>se_delta_elpd: Estimated standard error of delta_elpd</li>
<li>z: delta_elpd/se_delta_elpd. A z-score that can be compared to a normal distribution to test if the difference in model fit is significant.</li>
</ul>
<p>In this case we see that the full (causal) model is significantly better than the confounding model from the thrid line of the table. The <span class="math inline">\(z\)</span>-score is -6.53 corresponding to a p-value of 3.310^{-11}.</p>
<p>For each model (partial sharing and full) we can plot the posterior distributions of the parameters. Dotted lines show the prior distributions.</p>
<pre class="r"><code>plot(res$conf)</code></pre>
<p><img src="figure/ldl_cad.Rmd/plot1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot1-1">
Past versions of plot1-1.png
</button>
</p>
<div id="fig-plot1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/1753b227154a949d80ae0ece3d13ef18f92f29b7/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">1753b22</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/2eb09d8a6a0e5cb986f1ffd6a71a6e44ca2567b9/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">2eb09d8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/4a8f76ccd2554d1f46985507b41899ced9e02a3e/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">4a8f76c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/a34393dedf81051e37783951c9e3b74aba80de65/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">a34393d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/bbe49018f1c79ea1f2bfb1c1c4ac2228722dac60/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">bbe4901</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/73690eb831172fcd7130e9b4d2620f85b0c528c1/docs/figure/ldl_cad.Rmd/plot1-1.png" target="_blank">73690eb</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot(res$full)</code></pre>
<p><img src="figure/ldl_cad.Rmd/plot1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot1-2">
Past versions of plot1-2.png
</button>
</p>
<div id="fig-plot1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/1753b227154a949d80ae0ece3d13ef18f92f29b7/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">1753b22</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/2eb09d8a6a0e5cb986f1ffd6a71a6e44ca2567b9/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">2eb09d8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/4a8f76ccd2554d1f46985507b41899ced9e02a3e/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">4a8f76c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/a34393dedf81051e37783951c9e3b74aba80de65/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">a34393d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/bbe49018f1c79ea1f2bfb1c1c4ac2228722dac60/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">bbe4901</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/73690eb831172fcd7130e9b4d2620f85b0c528c1/docs/figure/ldl_cad.Rmd/plot1-2.png" target="_blank">73690eb</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The <code>summary</code> method will summarize the posterior medians and credible intervals.</p>
<pre class="r"><code>summary(res, ci_size=0.95)</code></pre>
<pre><code>p-value testing that causal model is a better fit:  3.3e-11 
Posterior medians and  95 % credible intervals:
     model              gamma               eta                  
[1,] &quot;Confounding Only&quot; NA                  &quot;0.38 (0.31, 0.45)&quot;  
[2,] &quot;Causal&quot;           &quot;0.34 (0.28, 0.39)&quot; &quot;-0.01 (-0.66, 0.54)&quot;
     q                  
[1,] &quot;0.78 (0.64, 0.88)&quot;
[2,] &quot;0.03 (0, 0.25)&quot;   </code></pre>
<p>The <code>plot</code> method applied to a <code>cause</code> object will arrange all of this information on one spread.</p>
<pre class="r"><code>plot(res)</code></pre>
<p><img src="figure/ldl_cad.Rmd/plot2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot2-1">
Past versions of plot2-1.png
</button>
</p>
<div id="fig-plot2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/1753b227154a949d80ae0ece3d13ef18f92f29b7/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">1753b22</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/2eb09d8a6a0e5cb986f1ffd6a71a6e44ca2567b9/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">2eb09d8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/4a8f76ccd2554d1f46985507b41899ced9e02a3e/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">4a8f76c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/a34393dedf81051e37783951c9e3b74aba80de65/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">a34393d</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/bbe49018f1c79ea1f2bfb1c1c4ac2228722dac60/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">bbe4901</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/73690eb831172fcd7130e9b4d2620f85b0c528c1/docs/figure/ldl_cad.Rmd/plot2-1.png" target="_blank">73690eb</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-10-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The <code>plot</code> method can also produce scatter plots of the data showing for each model, the probability that each variant is acting through the confounder and the contribution of each variant to the ELPD test statistic.</p>
<pre class="r"><code>plot(res, type=&quot;data&quot;)</code></pre>
<p><img src="figure/ldl_cad.Rmd/plot3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot3-1">
Past versions of plot3-1.png
</button>
</p>
<div id="fig-plot3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/1cc48608917d72a60217a93bfb658e467af6c3ed/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">1cc4860</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/2eb09d8a6a0e5cb986f1ffd6a71a6e44ca2567b9/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">2eb09d8</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jean997/cause/blob/4a8f76ccd2554d1f46985507b41899ced9e02a3e/docs/figure/ldl_cad.Rmd/plot3-1.png" target="_blank">4a8f76c</a>
</td>
<td>
Jean Morrison
</td>
<td>
2018-11-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] cause_0.2.0.0126 dplyr_0.8.3      readr_1.3.1     

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5     xfun_0.8             ashr_2.2-32         
 [4] purrr_0.3.2          lattice_0.20-38      colorspace_1.4-1    
 [7] vctrs_0.2.0          htmltools_0.3.6      loo_2.1.0           
[10] yaml_2.2.0           utf8_1.1.4           rlang_0.4.0         
[13] mixsqp_0.1-97        pillar_1.4.2         glue_1.3.1          
[16] matrixStats_0.54.0   foreach_1.4.4        stringr_1.4.0       
[19] munsell_0.5.0        gtable_0.3.0         workflowr_1.4.0.9000
[22] codetools_0.2-16     evaluate_0.14        labeling_0.3        
[25] knitr_1.23           doParallel_1.0.14    pscl_1.5.2          
[28] parallel_3.6.1       fansi_0.4.0          highr_0.8           
[31] Rcpp_1.0.1           backports_1.1.4      scales_1.0.0        
[34] RcppParallel_4.4.3   truncnorm_1.0-8      fs_1.3.1            
[37] gridExtra_2.3        ggplot2_3.2.0        hms_0.5.0           
[40] digest_0.6.20        stringi_1.4.3        numDeriv_2016.8-1.1 
[43] grid_3.6.1           rprojroot_1.3-2      cli_1.1.0           
[46] tools_3.6.1          magrittr_1.5         lazyeval_0.2.2      
[49] tibble_2.1.3         crayon_1.3.4         whisker_0.3-2       
[52] tidyr_0.8.3          pkgconfig_2.0.2      zeallot_0.1.0       
[55] MASS_7.3-51.4        Matrix_1.2-17        SQUAREM_2017.10-1   
[58] assertthat_0.2.1     rmarkdown_1.13       iterators_1.0.10    
[61] R6_2.4.0             intervals_0.15.1     git2r_0.26.1        
[64] compiler_3.6.1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
